chakra-cli doesn't find files referenced as custom paths in tsconfig.json

### Description

chakra-cli doesn't find files referenced as custom paths in tsconfig.json

### Link to Reproduction

N/A (Code Sandbox doesn't support running custom package.json scripts)

### Steps to reproduce

The error:

```bash
➜ chakra-cli tokens src/theme/index.ts

Chakra UI CLI  v1.5.0 by chakra UI
Generate theme typings for autocomplete


✖ An error occurred
Error: Cannot find module 'src/lib/guard-color-scheme'
Require stack:
- /.../src/theme/components/alert.ts
- /.../src/theme/index.ts
- /.../node_modules/@chakra-ui/cli/dist/scripts/read-theme-file.worker.js
```

File structure:

```bash
➜ tree src

src
├── index.ts
├── lib
│   └── guard-color-scheme.ts
└── theme
    ├── components
    │   ├── alert.ts
    ├── index.ts
```

```json
{
  "compilerOptions": {
    ...
    "paths": {
      "src/*": ["./*"]
    }
	...
  }
}
```

`alert.ts`:

```js
import guardColorScheme from 'src/lib/guard-color-scheme';
...
```

I would expect that `chakra-cli` correctly parses the project's `tsconfig.json` (or you can pass a path to a `tsconfig.json` as an argument) so that custom defined paths are correctly resolved.

### Chakra UI Version

1.6.6

### Browser

_No response_

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Version of `@chakra-ui/cli`: `1.5.0`

--- a/.changeset/lemon-beers-relax.md
+++ b/.changeset/lemon-beers-relax.md
@@ -1,0 +1,22 @@
+---
+"@chakra-ui/cli": patch
+---
+
+Fixed an issue where the CLI did not resolve custom tsconfig paths.
+
+🚨 Please note that only the first alias target from the string array will be
+resolved.
+
+```json5
+// tsconfig.json
+{
+  //...
+  compilerOptions: {
+    baseUrl: "src",
+    paths: {
+      "@alias/*": ["target/*"],
+      //           ^-- only the first target will be resolved
+    },
+  },
+}
+```

--- a/tooling/cli/package.json
+++ b/tooling/cli/package.json
@@ -39,11 +39,10 @@
     "prebuild": "rimraf dist",
     "start": "nodemon --watch src --exec 'yarn build; yarn dev' -e ts,tsx",
     "dev": "node bin/index.js tokens ../../website/theme.ts",
-    "build": "concurrently yarn:build:*",
     "test": "jest --env=jsdom --passWithNoTests",
     "lint": "concurrently yarn:lint:*",
     "version": "yarn build",
-    "build:cjs": "cross-env BABEL_ENV=cjs babel src --root-mode upward --extensions .ts,.tsx -d dist --source-maps",
+    "build": "cross-env BABEL_ENV=cjs babel src --root-mode upward --extensions .ts,.tsx -d dist --source-maps",
     "test:cov": "yarn test --coverage",
     "lint:src": "eslint src --ext .ts,.tsx --config ../../.eslintrc",
     "lint:types": "tsc --noEmit"
@@ -54,12 +53,15 @@
     "cli-handle-unhandled": "^1.1.1",
     "cli-welcome": "^2.2.2",
     "commander": "^6.2.1",
+    "module-alias": "^2.2.2",
     "ora": "^5.3.0",
     "regenerator-runtime": "^0.13.7",
     "ts-node": "9.1.1",
+    "tsconfig-paths": "^3.11.0",
     "update-notifier": "^5.0.1"
   },
   "devDependencies": {
+    "@types/module-alias": "^2.0.1",
     "@types/ora": "^3.2.0",
     "@types/update-notifier": "5.0.0"
   }

--- a/tooling/cli/src/scripts/read-theme-file.worker.ts
+++ b/tooling/cli/src/scripts/read-theme-file.worker.ts
@@ -1,7 +1,9 @@
 import "regenerator-runtime/runtime"
 import path from "path"
 import fs from "fs"
-import { register } from "ts-node"
+import * as tsNode from "ts-node"
+import * as tsConfigPaths from "tsconfig-paths"
+import moduleAlias from "module-alias"
 import { isObject } from "@chakra-ui/utils"
 import { createThemeTypingsInterface } from "../command/tokens/create-theme-typings-interface"
 import { themeKeyConfiguration } from "../command/tokens/config"
@@ -29,9 +31,49 @@ async function readTheme(themeFilePath: string) {
   const cwd = process.cwd()
   const absoluteThemePath = path.join(cwd, themeFilePath)
 
-  register({
-    project: path.join(__dirname, "..", "..", "bin", "tsconfig.json"),
-  })
+  const tsConfig = tsConfigPaths.loadConfig(absoluteThemePath)
+  if (tsConfig.resultType === "success") {
+    tsNode.register({
+      // use the TS projects own tsconfig file
+      project: tsConfig.configFileAbsolutePath,
+    })
+
+    /**
+     * Replace the module aliases in the transpiled code,
+     * because ts-node does not resolve them to relative require paths.
+     *
+     * 🚨 Note that only the first alias target will work
+     * @example tsconfig.json
+     * {
+     *   "baseUrl": "src",
+     *   "paths": {
+     *     "@alias/*": ["target/*"]
+     *   }
+     * }
+     */
+    const aliases = Object.keys(tsConfig.paths).reduce((acc, tsAlias) => {
+      // target/* -> target/
+      const firstTarget = tsConfig.paths[tsAlias][0].replace(/\*$/, "")
+      // @alias/* -> @alias
+      const jsAlias = tsAlias.replace(/\/\*$/, "")
+      // @alias = baseUrl/target/
+      acc[jsAlias] = path.join(tsConfig.absoluteBaseUrl, firstTarget)
+      return acc
+    }, {})
+    moduleAlias.addAliases(aliases)
+  } else {
+    // it is a JS project
+    const defaultProject = path.join(
+      __dirname,
+      "..",
+      "..",
+      "bin",
+      "tsconfig.json",
+    )
+    tsNode.register({
+      project: defaultProject,
+    })
+  }
 
   try {
     await fs.promises.stat(absoluteThemePath)

